src/components/Header/NavComponents/userprofile.tsx

```diff
@@ -30,6 +41,73 @@ export const UserProfile: React.FC<{}> = () => {
   const { data: agaveBalance } = useAmountAvailableToStake(address);
   const userBal = agaveBalance ? bigNumberToString(agaveBalance, 3) : "0";
 
+  const chainAddresses = useChainAddresses();
+
+  // internalAddressesPerNetworkId cannot be indexed for each chain, strigified and parsed back to index
+  var chains =  JSON.parse(JSON.stringify(internalAddressesPerNetworkId))
+
+  // Buttons to change to every available chain
+  let buttons : any[] = [];
+  Object.entries(internalAddressesPerNetworkId).forEach(([name, chain]) => {
+    buttons.push(
+      <Button 
+        bg={mode({ base: "primary.500", md: "primary.500" }, "primary.500")}
+        colorScheme="teal"
+        size="xl"
+        h="40px"
+        onClick={() => changeId(chain.chainId)}
+      >
+        {chain.chainName}
+      </Button>
+  )})
+
+  const currChainName : String = chains[chainAddresses?.chainId || 0]?.chainName
+
+  // Request wallet provider to change chain
+  // TODO: change it to enother file
+  function changeId(chainId : Number) {
+    const chain = chains[chainId.toString()]
+    try {
+      switch(chainId){
+        // If the chain is default to Metamask if will use wallet_switchEthereumChain
+        case 4:
+          window.ethereum.request({
+            id: 1,
+            jsonrpc: '2.0',
+            method: 'wallet_switchEthereumChain',
+            params: [
+              {
+                chainId: '0x4',
+              },
+            ],
+          })
+          break;
+        default:
+          window.ethereum.request({
+            id: 1,
+            jsonrpc: '2.0',
+            method: 'wallet_addEthereumChain',
+            params: [
+              {
+                chainId: "0x" + chain.chainId.toString(16),
+                chainName: chain.chainName,
+                rpcUrls: [chain.rpcUrl],
+                nativeCurrency: {
+                  name: chain.symbol,
+                  symbol: chain.symbol,
+                  decimals: 18,
+                },
+                blockExplorerUrls: [chain.explorer],
+              },
+            ],
+          })
+          break;
+      }
+    } catch {
+      console.error("Chain ID change failed")
```

Comment:
This probably can't happen, as the RPC calls are async- *unless* `request` is a synchronous call, at which point, these aren't even supported by most wallets anymore.

Reply:
Removed